src/app/qgsappbrowserproviders.cpp

```diff
@@ -1129,6 +1131,28 @@ void QgsBookmarksItemGuiProvider::populateContextMenu( QgsDataItem *item, QMenu
     menu->addAction( addBookmarkToGroup );
     menu->addSeparator();
 
+    // Rename bookmark group
+    QAction *renameBookmarkGroup = new QAction( tr( "Rename Bookmark Group…" ), menu );
+    connect( renameBookmarkGroup, &QAction::triggered, this, [ = ]
```

Comment:
```suggestion
    QAction *renameBookmarkGroup = new QAction( tr( "Rename Bookmark Group…" ), menu );
    const QString groupName = groupItem->group();
    connect( renameBookmarkGroup, &QAction::triggered, this, [groupName, manager]
```

It's potentially dangerous accessing groupItem directly in the slot here -- there's a small chance it will be deleted in between the menu showing and the user selecting the action from the menu.